### ISSUE TITLE
Typo in lint details

### ISSUE THREAD

**Issue by osa1** [dart-lang/site-www#4937]
### Page URL

https://dart.dev/tools/linter-rules#deprecated_member_use_from_same_package

### Describe the problem

In the sentence "Elements which are annotated with @deprecated should not be referenced ...", `@deprecated` should be `@Deprecated`, same as the examples after the sentence.

I was going to edit this myself but I can't figure out how.. The file https://github.com/dart-lang/site-www/blob/main/src/_data/linter_rules.json seems to be imported/generated, but it doesn't have any documentation. For future reference, how do I edit this file?

### Expected fix

`@deprecated` -> `@Deprecated`

**Comment by MaryaBelanger** [comment]
Good catch @osa1! I think this will need to be fixed in the [linter repo](https://github.com/dart-lang/linter/tree/main), and then maybe running a tool there generates the docs on this site. 

[This section of their contributing guide](https://github.com/dart-lang/linter/blob/main/CONTRIBUTING.md#mechanics) sounds like all you have to do is open a PR. I'm not sure how that generates to `linter-rules.md` here, but I'm sure they'll be able to explain once your PR is up (I'd like to know too, so I'll be keeping my eye on it :)). 

Please link it to this issue to close it, thank you!

**Comment by parlough** [comment]
Our system for this is going to change soon, but currently this page is generated following this process:

1. The descriptions in the linter rule implementation file (https://github.com/dart-lang/linter/tree/main/lib/src/rules)
2. Once that is merged, a `.json` file with information about each linter rule is generated and pushed to a different branch (https://github.com/dart-lang/linter/blob/gh-pages/lints/machine/rules.json)
3. We occasionally update our copy of that json file on dart.dev (https://github.com/dart-lang/site-www/blob/main/src/_data/linter_rules.json)
4. We use that data to generate the linter_rules page (https://github.com/dart-lang/site-www/blob/main/src/_includes/linter-rules-section.md)
